wordpress-mobile/WordPress-Android
# Fixes #9122: remote preview for pages

Fixes #9122

## Description

Before this patch we were only showing the local preview on pages (even on published post without modifications).

Before|After
-|-
![Screenshot_1563894218](https://user-images.githubusercontent.com/40213/61723202-00065900-ad6c-11e9-8c29-54d0ce87a696.png)|![Screenshot_1563894787](https://user-images.githubusercontent.com/40213/61723904-2f699580-ad6d-11e9-85c0-a8fe216d37c5.png)



For draft with modification, we were also showing the local preview:

Before|After
-|-
![Screenshot_1563894240](https://user-images.githubusercontent.com/40213/61723207-01d01c80-ad6c-11e9-915f-7050ce0cde4a.png) | ![Screenshot_1563894815](https://user-images.githubusercontent.com/40213/61723920-385a6700-ad6d-11e9-9722-75fe1d1049e8.png)



Notes;
- I removed `EditPostPreviewFragment` in this PR, because it wasn't used anymore. It was worth it's own PR 😬 .

### Test - published post

From the page list:

- On a published page
- Tap the "View" button from the 3 dots button
- Make sure the preview is displayed in the "preview" webview


### Test - published post with modifications

From the page list:

- On a published page with modifications
- Tap the "View" button from the 3 dots button
- Make sure the preview is generated and displayed in the "preview" webview

### Test - preview draft from the editor

- Open an existing draft
- From Overflow  menu, press "Preview"
- Make sure the preview is generated and displayed in the "preview" webview

### Test - preview draft with local changes from the editor

- Open an existing draft in the editor, modify it.
- From Overflow  menu, press "Preview"
- Make sure the preview is generated and displayed in the "preview" webview


### Test - new draft

- Create a new draft in the editor, add a title.
- From Overflow  menu, press "Preview"
- Make sure the preview is generated and displayed in the "preview" webview

## Known issues:

- Same as for posts, if the user edit concurrently edit drafts on Calypso (or another device), and preview it, the preview process will push the new version and could override some post edits silently.
- The "Saving / Generating" preview is not cancellable https://github.com/wordpress-mobile/WordPress-Android/issues/10141

## Update release notes:
